#!/bin/sh -e
## 02_aclocal_libs_fix.dpatch by Jordi Mallach <jordi@debian.org>
##
## DP: Don't modify LIBS if there's no need to. Patch from Josip Rodin.

if [ $# -lt 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch}"

case "$1" in
       -patch) patch $patch_opts -p1 < $0;;
       -unpatch) patch $patch_opts -p1 -R < $0;;
        *)
                echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
                exit 1;;
esac

exit 0

@DPATCH@
diff -urNad alsa-lib-1.0.12~/utils/alsa.m4 alsa-lib-1.0.12/utils/alsa.m4
--- alsa-lib-1.0.12~/utils/alsa.m4	2006-09-03 15:28:21.000000000 +0200
+++ alsa-lib-1.0.12/utils/alsa.m4	2006-09-03 15:27:35.000000000 +0200
@@ -43,6 +43,7 @@
 	CFLAGS="$CFLAGS -I$alsa_inc_prefix"
 fi
 AC_MSG_RESULT($ALSA_CFLAGS)
+CFLAGS="$alsa_save_CFLAGS"
 
 dnl add any special lib dirs
 AC_MSG_CHECKING(for ALSA LDFLAGS)
@@ -119,19 +120,15 @@
 )
 fi
 
+LDFLAGS="$alsa_save_LDFLAGS"
+LIBS="$alsa_save_LIBS"
+
 if test "x$alsa_found" = "xyes" ; then
    ifelse([$2], , :, [$2])
-   LIBS=`echo $LIBS | sed 's/-lasound//g'`
-   LIBS=`echo $LIBS | sed 's/  //'`
-   LIBS="-lasound $LIBS"
-fi
-if test "x$alsa_found" = "xno" ; then
-   ifelse([$3], , :, [$3])
-   CFLAGS="$alsa_save_CFLAGS"
-   LDFLAGS="$alsa_save_LDFLAGS"
-   LIBS="$alsa_save_LIBS"
+else
    ALSA_CFLAGS=""
    ALSA_LIBS=""
+   ifelse([$3], , :, [$3])
 fi
 
 dnl That should be it.  Now just export out symbols:
